Common++/header/IpAddresses.h

```diff
@@ -0,0 +1,551 @@
+	 * Another example: if subnet is 10.1.X.X, subnet mask is 255.0.0.0 and address is 11.1.1.9 then the method will return false
+	 * @param[in] addr The address to be checked
+	 * @param[in] subnet The subnet to be verified. Notice it's an IPv4Address type, so subnets with don't-cares (like 10.0.0.X) must have some number
+	 * (it'll be ignored if subnet mask is correct)
+	 * @param[in] subnetMask The subnet mask to compare the address with the subnet
+	 */
+	bool matchSubnet(const IPv4Address& addr, const IPv4Address& subnet, const IPv4Address& subnetMask);
+
+
+
+
+	/**
+	 * @class IPv6Address
+	 * Represents an IPv6 address (of type xxxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx).
+	 */
+	class IPv6Address
+	{
+	public:
+		/**
+		 * A default constructor that creates an instance of the class with unspecified/zero address
+		 */
+		IPv6Address() { memset(m_Bytes, 0, sizeof(m_Bytes)); }
+
+		/**
+		 * A constructor that creates an instance of the class out of 16-byte array.
+		 * @param[in] bytes The address as 16-byte array in network byte order
+		 */
+		IPv6Address(const uint8_t bytes[16]) { memcpy(m_Bytes, bytes, sizeof(m_Bytes)); }
+
+		/**
+		 * Returns a pointer to 16-byte array representing the IPv6 address
+		 */
+		const uint8_t* toBytes() const { return m_Bytes; }
+
+		/**
+		 * Returns a std::string representation of the address
+		 * @return A string representation of the address
+		 */
+		std::string toString() const;
+
+		/**
+		 * Determine whether the address is unspecified
+		 */
+		bool isUnspecified() const { return *this == Zero; }
+
+		/**
+		 * Overload of the equal-to operator
+		 */
+		bool operator==(const IPv6Address& rhs) const { return memcmp(toBytes(), rhs.toBytes(), sizeof(m_Bytes)) == 0; }
+
+		/**
+		 * Overload of the not-equal-to operator
+		 */
+		bool operator!=(const IPv6Address &rhs) const { return !(*this == rhs); }
+
+
+		// Following methods are placed for backward compatibility and will be deleted in the future
+
+		/**
+		 * Clone the object
+		 * Notice this method is deprecated and will be deleted in the future
+		 * @return A newly allocated instance which is a clone of the current instance
+		 */
+		#if __cplusplus > 201402L || _MSC_VER >= 1900
+		[[deprecated("This method is now unnecessary and added for backward compatibility. It will be deleted in the future")]]
+		#endif
+		IPv6Address *clone() const { return new IPv6Address(*this); }
+
+		/**
+		 * Returns a in6_addr struct pointer representing the IPv6 address
+		 * Notice this method is deprecated and will be deleted in the future. Prefer toBytes method
+		 * @return a in6_addr struct pointer representing the IPv6 address
+		 */
+		#if __cplusplus > 201402L || _MSC_VER >= 1900
+		[[deprecated("This method is replaced by toBytes() method. It will be deleted in the future")]]
+		#endif
+		const in6_addr* toIn6Addr() const { return reinterpret_cast<const in6_addr*>(m_Bytes); }
```

Comment:
why deprecating this method? It might be useful for people who are working with the `in6_addr` struct